string_manipulation.rb

```diff
@@ -1,30 +1,133 @@
+# Helpers
+def swap_in_place(string, left, right) #time complexity = O(c), space = O(c) (temp_var)
+  temp_var = string[left]
+  string[left] = string[right]
+  string[right] = temp_var
+end
+
+
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  puts "NOT IMPLEMENTED"
+  return my_string if my_string.length < 2
+
+  i = 0
+
+  while i < my_string.length/2
+    swap_in_place(my_string, i, my_string.length-i-1)
+    i +=1
+  end
+
+  return my_string
 end
 
 # A method to reverse each word in a sentence, in place.
-def reverse_words(my_words)
-  puts "NOT IMPLEMENTED"
+def reverse_words(my_words) #with string_reverse

```

Comment:
The time complexity of this would be O(n). For each word, you go through each character at the most 2 times i.e. 2 * n. This is true if there is one word, two words or multiple words. 
Space complexity, as you called out would be O(1).